src/NexusMods.App.UI/Pages/MyGames/MyGamesViewModel.cs

```diff
@@ -206,9 +232,17 @@ private OneOf<None, CreateLoadoutJob, UnmanageGameJob> GetJobRunningForGameInsta
         return OneOf<None, CreateLoadoutJob, UnmanageGameJob>.FromT0(new None());
     }
 
-    private async Task RemoveAllLoadouts(GameInstallation installation)
+    private async Task RemoveGame(GameInstallation installation, bool shouldDeleteDownloads, LibraryFile.ReadOnly[] filesToDelete, CollectionMetadata.ReadOnly[] collections)
     {
         await installation.GetGame().Synchronizer.UnManage(installation);
+
+        if (!shouldDeleteDownloads) return;
+        await _libraryService.RemoveItems(filesToDelete.Select(file => file.AsLibraryItem()));
+
+        foreach (var collection in collections)
+        {
+            await _collectionDownloader.DeleteCollection(collection);
+        }
```

Comment:
I guess this is happening outside the unmanage Job then, should this be part of it instead?
Otherwise the app will say that unmanage has finished and the user might close the app while it is still deleting downloads and collections

Reply:
The widget will stay on the `RemovingGame` state until this task completes. I didn't want to put this stuff into the loadout synchronizer because our abstractions make this a complete mess.

Reply:
I guess we would have the problem we had before of switching away and back and the state being reset.
I guess it would literally be simpler to show a modal with a spinner while the task is running, so the user can't navigate away.